.github/workflows/release.yml

```
@@ -0,0 +1,56 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  HUSKY: 0
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [16, 18, 20]
+    name: Test on ${{ matrix.os }} with node v${{ matrix.node-version }}
+    steps:
+    - name: Setup Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        check-latest: true
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Install dependencies
+      run: npm install
+    - name: Run tests
+      run: npm test
```

I don't think we need to test first if we're running on dispatches against main, what do you think?

We should be in a passing state since you can only merge on passing tests, but I suppose there's a nonzero chance something happens.

I'll leave it up to you, but if we keep it:
```suggestion
  test:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
        node-version: [16, 18, 20]
    name: Test on ${{ matrix.os }} with node v${{ matrix.node-version }}
    steps:
    - name: Setup Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v4
      with:
        node-version: ${{ matrix.node-version }}
        check-latest: true
    - name: Checkout repository
      uses: actions/checkout@v4
    - name: Install dependencies
      run: npm install
    - name: Run tests
      run: npm test
```

---

once again, a good suggestion - thanks. i don't think we need to test again. i will remove the redundant tests.